bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -138,7 +138,7 @@ long getEntryId() {
     void sendWriteRequest(int bookieIndex) {
         int flags = isRecoveryAdd ? FLAG_RECOVERY_ADD | FLAG_HIGH_PRIORITY : FLAG_NONE;
 
-        lh.bk.getBookieClient().addEntry(lh.getLedgerMetadata().currentEnsemble.get(bookieIndex),
+        lh.bk.getBookieClient().addEntry(lh.getCurrentEnsemble().get(bookieIndex),
```

Comment:
Sample the list outside of this method ? in the loop I mean

Reply:
done